pom.xml

```diff
@@ -390,2363 +341,2523 @@
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.apache.thrift</groupId>
+                        <artifactId>libfb303</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.slf4j</groupId>
+                        <artifactId>slf4j-api</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.slf4j</groupId>
+                        <artifactId>slf4j-log4j12</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>log4j</groupId>
+                        <artifactId>log4j</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>commons-logging</groupId>
+                        <artifactId>commons-logging</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.codehaus.groovy</groupId>
+                        <artifactId>groovy-all</artifactId>
+                    </exclusion>
+                    <!-- Begin of Hive 2.3 exclusion -->
+                    <exclusion>
+                        <groupId>${hive.group}</groupId>
+                        <artifactId>hive-service-rpc</artifactId>
+                    </exclusion>
+                    <!-- parquet-hadoop-bundle:1.8.1 conflict with 1.12.0 -->
+                    <exclusion>
+                        <groupId>org.apache.parquet</groupId>
+                        <artifactId>parquet-hadoop-bundle</artifactId>
+                    </exclusion>
+                    <!-- Do not need Jasper, see HIVE-19799 -->
+                    <exclusion>
+                        <groupId>tomcat</groupId>
+                        <artifactId>jasper-compiler</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>tomcat</groupId>
+                        <artifactId>jasper-runtime</artifactId>
+                    </exclusion>
+                    <!-- End of Hive 2.3 exclusion -->
+                </exclusions>
+            </dependency>
 
-			<dependency>
-				<groupId>net.sf.jpam</groupId>
-				<artifactId>jpam</artifactId>
-				<scope>${hive.deps.scope}</scope>
-				<version>${jpam.version}</version>
-				<exclusions>
-					<exclusion>
-						<groupId>javax.servlet</groupId>
-						<artifactId>servlet-api</artifactId>
-					</exclusion>
-				</exclusions>
-			</dependency>
-
-			<!-- hive shims pulls in hive 0.23 and a transitive dependency of the Hadoop version
+            <dependency>
+                <groupId>${hive.group}</groupId>
+                <artifactId>hive-service-rpc</artifactId>
+                <version>3.1.2</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>*</groupId>
```

Comment:
what is left of this dependency if everything is excluded?...

Reply:
I would say the dependency itself remains, but no sub-dependencies...